src/csharp/Microsoft.Spark.Worker/Command/SqlCommandExecutor.cs

```diff
@@ -51,18 +53,51 @@ internal abstract class SqlCommandExecutor
                 throw new ArgumentException("Unexpected serialization mode found.");
             }
 
+            bool useDataFrameCommandExecutor = false;
+            foreach (SqlCommand command in commands)
+            {
+                WorkerFunction workerFunc = command.WorkerFunction;
+                DataFrameWorkerFunction dataFrameWorkerFunction = workerFunc as DataFrameWorkerFunction;
```

Comment:
I want to get feedback about this `if` pattern first. If it's acceptable, then I can address a bug where a user could create a `CommandPayload` defined with both `DataFrameWorkerFunction` and `ArrowSqlWorkerFunction`.

Reply:
Similar pattern in `CommandProcessor.cs`

Reply:
I wonder if this wouldn't be better lower in the stack? For example, using the `ArrowSqlCommandExecutor` for both cases, and then in `ICommandRunner commandRunner = CreateCommandRunner(commands);` you can look at the command and decide whether to go the "data frame" route or the "arrow array" route.

Note: if you keep the current design, I would refactor this chunk out into a separate method.

Reply:
Done. I made a new `ArrowOrDataFrameSqlCommandExecutor` class that internally chooses which SqlCommandExecutor to call based on the type of the `workerFunction`.